pkg/operation/botanist/monitoring_test.go

```diff
@@ -280,5 +286,21 @@ var _ = Describe("Monitoring", func() {
 			Expect(secret.Labels).To(HaveKeyWithValue("gardener.cloud/role", "monitoring"))
 			Expect(secret.Data).To(And(HaveKey("username"), HaveKey("password"), HaveKey("auth"), HaveKey("basic_auth.csv")))
 		})
+
+		It("should cleanup the secrets when shoot purpose is changed", func() {
+			defer test.WithVar(&ChartsPath, filepath.Join("..", "..", "..", "charts"))()
+
+			Expect(fakeGardenClient.Get(ctx, kutil.Key(projectNamespace, shootName+".monitoring"), &corev1.Secret{})).To(BeNotFoundError())
+
+			Expect(botanist.DeploySeedGrafana(ctx)).To(Succeed())
+			Expect(fakeGardenClient.Get(ctx, kutil.Key(projectNamespace, shootName+".monitoring"), &corev1.Secret{})).To(Succeed())
+			Expect(*botanist.Shoot.GetInfo().Spec.Purpose == shootPurposeEvaluation).To(BeTrue())
+
+			botanist.Shoot.Purpose = shootPurposeTesting
+			Expect(botanist.DeploySeedGrafana(ctx)).To(Succeed())
+			Expect(botanist.Shoot.Purpose == shootPurposeTesting).To(BeTrue())
```

Comment:
Drop this line, it doesn't yield any benefit, does it?

Reply:
ohh.. sorry. I forgot to remove it.